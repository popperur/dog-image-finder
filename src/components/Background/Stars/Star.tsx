import { StarHolder } from "components/Background/Stars/styles.tsx";

interface StarProps {
  top: number;
  left: number;
  width: number;
  height: number;
  animationDelay: number;
}

function Star({ top, left, width, height, animationDelay }: StarProps) {
  const stylesArray = [
    `top: ${top}%`,
    `left: ${left}%`,
    `width: ${width}px`,
    `height: ${height}px`,
    `animation-delay: ${animationDelay}s`,
  ];

  return <StarHolder $styles={stylesArray} />;
}

export default Star;
