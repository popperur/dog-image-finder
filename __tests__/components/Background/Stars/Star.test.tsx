import "jest-styled-components";
import { render, RenderResult } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import Star from "components/Background/Stars/Star";

describe("Star component", () => {
  let rendered: RenderResult;

  beforeEach(() => {
    rendered = render(
      <Star top={0} left={20} width={40} height={60} animationDelay={1.5} />,
    );
  });

  it("matches the snapshot", () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  it("adds the styles to the styled component", () => {
    const { container } = rendered;

    const starHolder = container.firstChild;

    expect(starHolder).toHaveStyleRule("top", "0%");
    expect(starHolder).toHaveStyleRule("left", "20%");
    expect(starHolder).toHaveStyleRule("width", "40px");
    expect(starHolder).toHaveStyleRule("height", "60px");
    expect(starHolder).toHaveStyleRule("animation-delay", "1.5s");
  });
});
