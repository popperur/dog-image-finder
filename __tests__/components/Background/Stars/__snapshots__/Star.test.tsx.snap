// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Star component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 0%;
  left: 20%;
  width: 40px;
  height: 60px;
  animation-delay: 1.5s;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
